src/redux/books.ts

```diff
@@ -1,7 +1,61 @@
-import { useSelector } from "react-redux";
-import { RootState } from "./store";
+import { PayloadAction, createAsyncThunk, createSlice } from "@reduxjs/toolkit";
+import axios from "axios";
 
-export const useBooks = () => {
-    return useSelector((state: RootState) => state.libraryDetails.books);
-  };
+interface InitialState {
+  loading: boolean,
+  books: Book[],
+  error: string
+};
 
+interface Book {
+  id: number, 
+  type: string, 
+  attributes: {
+   isbn: string, 
+   book_image: string, 
+   description: string, 
+   title: string, 
+   author: string, 
+   genre: string,
+   library_id: number
+  }
+};
+
+interface BookDetailsResponse {
+  data: Book[]
+};
+
+const initialState: InitialState = {
+  loading: false,
+  books: [],
+  error: ""
+};
+
+export const fetchSpecificBooks = createAsyncThunk('bookDetails/fetchSpecificBooks', (libraryId: number) => {
+  return axios 
+    .get(`https://book-haven-be-29aa9bd8a3c7.herokuapp.com/api/v0/libraries/${libraryId}/books`)
+    .then(response => response.data)
+});
+
+export const bookDetailsSlice = createSlice({
+  name: "bookDetails",
+  initialState,
+  reducers: {}, // TO CHECK: do addBook and removeBook need to be reducers here for buttons?
```

Comment:
We can add reducers here when we consolidate libraryDetailsView with BookDetails